src/error.F90

```diff
@@ -18,24 +18,36 @@ module error
 ! stream.
 !===============================================================================
 
-  subroutine warning(message)
+  subroutine warning(message_in, deprecation)
 
-    character(*) :: message
-
-    integer :: i_start   ! starting position
-    integer :: i_end     ! ending position
-    integer :: line_wrap ! length of line
-    integer :: length    ! length of message
-    integer :: indent    ! length of indentation
+    character(*)                  :: message_in  ! printed with the warning
+    logical, optional, intent(in) :: deprecation ! is a depreciation warning
 
-    ! Write warning at beginning
-    write(ERROR_UNIT, fmt='(1X,A)', advance='no') 'WARNING: '
+    integer                   :: i_start      ! starting position
+    integer                   :: i_end        ! ending position
+    integer                   :: line_wrap    ! length of line
+    integer                   :: length       ! length of message
+    integer                   :: indent       ! length of indentation
+    character(:), allocatable :: message      ! input message with a prefix
```

Comment:
I don't think reworking `warning` is worth the effort right now.  If it's okay with you, I'd rather just revert to the develop version of `warning`.